libraries/AP_HAL_ChibiOS/hwdef/speedybeef4v3/hwdef.dat

```diff
@@ -0,0 +1,155 @@
+# tested on the Speedybee F4 v3 board
+# with thanks to betaflight for pinout
+
+# MCU class and specific type
+MCU STM32F4xx STM32F405xx
+
+# board ID for firmware load
+APJ_BOARD_ID 1082
+
+# crystal frequency
+OSCILLATOR_HZ 8000000
+
+define STM32_ST_USE_TIMER 5
+
+FLASH_SIZE_KB 1024
+
+# only one I2C bus
+I2C_ORDER I2C2
+
+# order of UARTs (and USB)
+SERIAL_ORDER OTG1 USART1 USART2 USART3 UART4 UART5 USART6
+
+# LEDs
+PC8 LED_BLUE OUTPUT LOW GPIO(0)
+
+define HAL_GPIO_A_LED_PIN 0
+
+# buzzer
+PC5 BUZZER OUTPUT GPIO(80) LOW
+define HAL_BUZZER_PIN 80
+define HAL_BUZZER_ON 1
+define HAL_BUZZER_OFF 0
+
+# spi1 bus for IMU
+PA5 SPI1_SCK SPI1
+PA6 SPI1_MISO SPI1
+PA7 SPI1_MOSI SPI1
+PA4 BMI270_CS CS
+
+# spi2 for OSD and sdcard
+PB13 SPI2_SCK SPI2
+PB14 SPI2_MISO SPI2
+PB15 SPI2_MOSI SPI2
+PB12 MAX7456_CS CS
+PA15 SDCARD_CS CS
+
+# only one I2C bus in normal config
+PB10 I2C2_SCL I2C2
+PB11 I2C2_SDA I2C2
+
+# analog pins
+PC0 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+PC1 BATT_CURRENT_SENS ADC1 SCALE(1)
+PC2 RSSI_ADC_PIN ADC1 SCALE(1)
+
+# define default battery setup
+define HAL_BATT_MONITOR_DEFAULT 4
+define HAL_BATT_VOLT_PIN 10
+define HAL_BATT_CURR_PIN 11
+define HAL_BATT_VOLT_SCALE 11.2 # matched to ESC output
+define HAL_BATT_CURR_SCALE 52.7 # appropriate for a T-Motor F55A Pro II
+
+# analog rssi pin
+define BOARD_RSSI_ANA_PIN 12
+
+# USART1 (DJI / VTX)
+PA9 USART1_TX USART1
+PA10 USART1_RX USART1
+define HAL_SERIAL1_PROTOCOL SerialProtocol_DJI_FPV
+
+# USART2 (RCIN)
+PA2 USART2_TX USART2
+PA3 USART2_RX USART2
+define HAL_SERIAL2_PROTOCOL SerialProtocol_RCIN
+
+# USART3 (CAM)
+PC10 USART3_TX USART3 NODMA
+PC11 USART3_RX USART3 NODMA
+
+# UART4 (Bluetooth)
+PA0 UART4_TX UART4 NODMA
+PA1 UART4_RX UART4 NODMA
+
+# UART5 (ESC Telemetry)
+PC12 UART5_TX UART5 NODMA
+PD2  UART5_RX UART5 NODMA
+define HAL_SERIAL5_PROTOCOL SerialProtocol_ESCTelemetry
+define HAL_SERIAL5_BAUD 19200
+
+# UART6 (GPS)
+PC6 USART6_TX USART6
+PC7 USART6_RX USART6
+define HAL_SERIAL6_PROTOCOL SerialProtocol_GPS
+define HAL_SERIAL6_BAUD AP_SERIALMANAGER_GPS_BAUD
+
+# PA10 IO-debug-console
+PA11 OTG_FS_DM OTG1
+PA12 OTG_FS_DP OTG1
+
+# PWM out pins. Note that channel order follows the ArduPilot motor
+# order conventions
+PB6  TIM4_CH1 TIM4 PWM(1) GPIO(50) BIDIR
+PB7  TIM4_CH2 TIM4 PWM(2) GPIO(51)
+PB8  TIM4_CH3 TIM4 PWM(3) GPIO(52) BIDIR    # DMA channel shared with I2C2_TX, beware - no alternatives
+PB9  TIM4_CH4 TIM4 PWM(4) GPIO(53)
+PB0  TIM3_CH3 TIM3 PWM(5) GPIO(54)
+PB1  TIM3_CH4 TIM3 PWM(6) GPIO(55)
+PB5  TIM3_CH2 TIM3 PWM(7) GPIO(56)
+PB4  TIM3_CH1 TIM3 PWM(8) GPIO(57)
+PC9  TIM8_CH4 TIM8 PWM(9) GPIO(58) # LED
+
+# OSD pad for TTL based OSD control, not supported by AP
+PB3 CAM_C OUTPUT LOW GPIO(83)        # labelled as "CC"
+
+define HAL_STORAGE_SIZE 15360
+STORAGE_FLASH_PAGE 1
+
+# reserve 16k for bootloader and 32k for flash storage
+FLASH_RESERVE_START_KB 48
+
+# one IMU
+IMU BMI270 SPI:bmi270 ROTATION_ROLL_180
+
+# probe for a DPS280, but also look for other baros
```

Comment:
should not have baro probe and remove all other backends for space saving as we recently did for other 1MB boards

Reply:
Fixed